src/main/java/frc/team166/robot/subsystems/Lift.java

```diff
@@ -0,0 +1,197 @@
+shift into low or high gear 
+read top limit switch 
+read bottom limit switch 
+read other limit switches...? 
+ go to specified height
+
+Commands 
+go to specified height 
+make the robot climb 
+raise lift 
+lower lift 
+*/
+
+package frc.team166.robot.subsystems;
+
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.Preferences;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj.DoubleSolenoid.Value;
+import edu.wpi.first.wpilibj.command.Command;
+import edu.wpi.first.wpilibj.command.PIDSubsystem;
+import edu.wpi.first.wpilibj.command.Subsystem;
+import frc.team166.chopshoplib.commands.ActionCommand;
+import frc.team166.chopshoplib.commands.CommandChain;
+import frc.team166.chopshoplib.commands.SubsystemCommand;
+import frc.team166.robot.Robot;
+import frc.team166.robot.RobotMap;
+import frc.team166.robot.RobotMap.Encoders;
+import edu.wpi.first.wpilibj.DigitalInput;
+
+public class Lift extends PIDSubsystem {
+    // Defines Limit Switches (Digital imputs)
+    DigitalInput bottomLimitSwitch = new DigitalInput(RobotMap.DigitalInputs.LIFT_LIMIT_SWITCH_BOTTOM);
+    DigitalInput topLimitSwitch = new DigitalInput(RobotMap.DigitalInputs.LIFT_LIMIT_SWITCH_TOP);
+    // Defines Encoders 
+    Encoder liftEncoder = new Encoder(RobotMap.Encoders.LIFT_A, RobotMap.Encoders.LIFT_B);
+    // TODO we still need to calculate this
+    private final double encoderDistancePerTick = 0.01;
+    // Defines Motors 
+    WPI_TalonSRX liftMotorA = new WPI_TalonSRX(RobotMap.CAN.LIFT_MOTOR_A);
+    WPI_TalonSRX liftMotorB = new WPI_TalonSRX(RobotMap.CAN.LIFT_MOTOR_B);
+    // Defines the previus motors as one motor 
+    SpeedControllerGroup liftDrive = new SpeedControllerGroup(liftMotorA, liftMotorB);
+    DoubleSolenoid liftTransmission = new DoubleSolenoid(RobotMap.Solenoids.LIFT_TRANSMISSION_A,
+            RobotMap.Solenoids.LIFT_TRANSMISSION_B);
+
+    final static double kP = Preferences.getInstance().getDouble(RobotMap.Preferences.K_P, 1);
+    final static double kI = Preferences.getInstance().getDouble(RobotMap.Preferences.K_I, 1);
+    final static double kD = Preferences.getInstance().getDouble(RobotMap.Preferences.K_D, 1);
+    final static double kF = Preferences.getInstance().getDouble(RobotMap.Preferences.K_F, 1);
+
+    //enumerator that will be pulled from for the GoToHeight Command
+    public enum LiftHeight {
+        //will be changed
+        kFloor(0), kSwitch(1), kPortal(2), kIntake(3), kScaleLow(4), kScaleHigh(5), kClimb(6), kMaxHeight(7);
+
+        @SuppressWarnings("MemberName")
+        private double value;
+
+        //
+        LiftHeight(double value) {
+            this.value = value;
+        }
+
+        public double get() {
+            return value;
+        }
+    }
+
+    public Lift() {
+        super("Lift", kP, kI, kD, kF);
+        //creates a child for the encoders
```

Comment:
We should add two more things here to further set up the PID stuff...

setAbsoluteTolerance(0.05);  where the 0.05 is 5% of error we can tolerate

setOutputRange(downMaxSpeed, upMaxSpeed); where those those values are the fastest we want the robot to go when moving up or down (these should be made as preferences! use a default value of 1 for now)